src/multirust-cli/multirust_mode.rs

```diff
@@ -326,7 +326,7 @@ fn add_target(cfg: &Cfg, m: &ArgMatches) -> Result<()> {
         pkg: "rust-std".to_string(),
         target: TargetTriple::from_str(target),
     };
-    try!(toolchain.add_component(new_component));
+    try!(toolchain.add_component(new_component, &cfg));
```

Comment:
Ok, after looking at `telemetry` it seems the reason `Cfg` needs to be passed around is to get the value of `cfg.multirust_dir`. To refactor this so `Toolchain` doesn't need to be passed `Cfgs` I suggest the following:

* Add a `telemetry::Telemetry` type that contains a `telemetry_dir` field. When `Toolchain::new` builds itself from a `Cfg`, also build a `Telemetry` type and stash it in `Toolchain`. Then convert `log_telemetry` function to a `Telemetry::log` method.